cmd/execution/main.go

```diff
@@ -564,6 +570,83 @@ func main() {
 		}).Run()
 }
 
+// Get and log the epoch counters from the protocol state and from the FlowEpoch smart contract.
+func logEpochNumber(vm *fvm.VirtualMachine, vmCtx fvm.Context, executionState state.ExecutionState, node *cmd.NodeConfig, log zerolog.Logger) {
```

Comment:
It's better to make a function that simply reads the epoch counter from the smart contract.
it should either return
- (epochContractAddress, epochCounter, nil)  // if was able to read the epoch counter successfully
- ("", 0, err) // or run into an exception
```suggestion
func readEpochCounterFromSmartContract(vm *fvm.VirtualMachine, vmCtx fvm.Context, executionState state.ExecutionState) (string, uint64, error) {
```

Reply:
refactored a bit